tests/chainer_tests/links_tests/connection_tests/test_lstm.py

```diff
@@ -57,4 +57,94 @@ def test_forward_gpu(self):
         self.check_forward(cuda.to_gpu(self.x))
 
 
+class TestLSSTMRestState(unittest.TestCase):
+
+    def setUp(self):
+        self.link = links.LSTM(5, 7)
+        self.x = chainer.Variable(
+            numpy.random.uniform(-1, 1, (3, 5)).astype(numpy.float32))
+
+    def check_state(self):
+        self.assertIsNone(self.link.c)
+        self.assertIsNone(self.link.h)
+        self.link(self.x)
+        self.assertIsNotNone(self.link.c)
+        self.assertIsNotNone(self.link.h)
+
+    def test_state_cpu(self):
+        self.check_state()
+
+    @attr.gpu
+    def test_state_gpu(self):
+        self.link.to_gpu()
+        self.x.to_gpu()
+        self.check_state()
+
+    def check_reset_state(self):
+        self.link(self.x)
+        self.link.reset_state()
+        self.assertIsNone(self.link.c)
+        self.assertIsNone(self.link.h)
+
+    def test_reset_state_cpu(self):
+        self.check_reset_state()
+
+    @attr.gpu
+    def test_reset_state_gpu(self):
+        self.link.to_gpu()
+        self.x.to_gpu()
+        self.check_reset_state()
+
+
+class TestLSTMToCPUToGPU(unittest.TestCase):
+
+    def setUp(self):
+        self.link = links.LSTM(5, 7)
+        self.x = chainer.Variable(
+            numpy.random.uniform(-1, 1, (3, 5)).astype(numpy.float32))
+
+    def check_to_cpu(self, s):
+        self.link.to_cpu()
+        self.assertIsInstance(s.data, self.link.xp.ndarray)
+        self.link.to_cpu()
```

Comment:
First assert tries to check that `to_cpu` forcibly moves the state to CPU even if link is on CPU or GPU. Second assert tries to check the idempotency of `to_cpu`.